module/DTO/questionsDTO.js

```diff
@@ -0,0 +1,541 @@
+
+        const cacheData = await getKey(referenceQuestionSetId).catch((err) => {
+          console.log("Error in getting data from Redis:", err);
+        });
+        //if cache data present
+        if (cacheData) {
+          resolve({
+            success: true,
+            message: messageConstants.apiResponses.EVIDENCE_FETCHED,
+            data: JSON.parse(cacheData),
+          });
+        } else {
+          // get data from the creation portal
+          const res = await readQuestionSet(referenceQuestionSetId).catch(
+            (err) => {
+              console.log("Error", err?.response?.data);
+              reject({
+                success: false,
+                message: err?.response?.data,
+                data: false,
+              });
+            }
+          );
+
+          if (res.body.responseCode !== httpStatusCode.ok.code) {
+            return reject({
+              message: res.params.errmsg,
+              status: httpStatusCode.bad_request.status,
+            })
+          }
+
+          const questionSetHierarchy =
+            res?.result?.questionSet || res?.result?.questionset;
+
+          const migratedCriteriaQuestions =
+            questionSetHierarchy?.children || [];
+          const evidences = await this.questionEvidences(
+            migratedCriteriaQuestions,
+            submissionDocumentCriterias,
+            isPageQuestionsRequired
+          )?.data;
+
+          assessmentTemplate.assessment.evidences[0].name = capitalize(
+            solutionDocument?.type
+          );
+          assessmentTemplate.assessment.evidences[0].sections[0].name = `${capitalize(
+            solutionDocument?.type
+          )} Questions`;
+
+          assessmentTemplate.assessment.evidences[0].sections[0].questions =
+            evidences?.evidenceSections || [];
+          assessmentTemplate.assessment.evidences[0].sections[0].code = "SQ";
+          assessmentTemplate.assessment.evidences[0].code =
+            assessmentTemplate.assessment.evidences[0].externalId = "SF";
+          assessmentTemplate.assessment.evidences[0].description =
+            questionSetHierarchy?.description || "";
+
+          await setKey(
+            solutionDocument.referenceQuestionSetId,
+            {
+              ...evidences,
+              evidences: assessmentTemplate.assessment.evidences,
+            },
+            cacheTtl
+          );
+
+          resolve({
+            success: true,
+            message: messageConstants.apiResponses.EVIDENCE_FETCHED,
+            data: {
+              ...evidences,
+              evidences: assessmentTemplate.assessment.evidences,
+            },
+          });
+        }
+      } catch (error) {
+        return reject({
+          success: false,
+          message: error.message,
+          data: false,
+        });
+      }
+    });
+  }
+
+  /**
+   * Processes criteria questions and generates evidence sections.
+   *
```

Comment:
@jayesh9747 please correct the function signtaure in entire code which u have witten